django/db/models/sql/query.py

```diff
@@ -1770,7 +1767,7 @@ def split_exclude(self, filter_expr, can_reuse, names_with_path):
         # Generate the inner query.
         query = Query(self.model)
         query._filtered_relations = self._filtered_relations
-        query.add_filter(filter_expr)
+        query.add_q(Q(filter_expr))
```

Comment:
For readability I'd probably move `filter_expr` here :thinking:, but it's a separate issue:
```python
        # Generate the inner query.
        query = Query(self.model)
        query._filtered_relations = self._filtered_relations
        query.clear_ordering(force=True)
        filter_lhs, filter_rhs = filter_expr
        if isinstance(filter_rhs, OuterRef):
            query.add_q(Q(filter_lhs, OuterRef(filter_rhs))
        elif isinstance(filter_rhs, F):
            query.add_q(Q(filter_lhs, OuterRef(filter_rhs.name))
```

Reply:
I've done this, but a slight variant therein - re-using the restored `.add_filter(*filter_expr)` (because there's no `else` condition we need to fall back to re-consuming the original lhs & rhs if it's neither an `F` or an `OuterRef`)

Let me know if meets your criteria or not on that basis?

Reply:
We only change `filter_rhs` so I simplified it a bit.